### PR TITLE
Clarify visibility of protected methods

### PR DESCRIPTION
**Pull request by straight-shoota**
Superseeds #236

/cc @philxyz

### CODE REVIEW

**Comment by j8r** [comment]
A `module` can be added in examples.

**Comment by straight-shoota** [comment]
I fear the example is already too big. Besides module there could also be struct. I think it's fine as it is.

File: syntax_and_semantics/visibility.md [hunk 1]
```diff
@@ -118,16 +118,45 @@ end
...
+  def instance_method
+    Parent.protected_method # OK
+  end
+  
+  def self.class_method
+    Parent.protected_method # OK
+  end
+end
+
+class Parent::Sub
```
Comment (j8r) [inline syntax_and_semantics/visibility.md]: I mean replacing one of this classes by a module. There are already 3 classes

Comment (j8r) [inline syntax_and_semantics/visibility.md]: I mean replacing one of this classes by a module. There are already 3 classes

Comment (j8r) [reply syntax_and_semantics/visibility.md]: You know it, but new comers won't know systematically.

Comment (straight-shoota) [reply syntax_and_semantics/visibility.md]: They won't learn it either when this line is changed to `module`. On the contrary, it might create confusion.

**Pull request merged**